docker-app/qfieldcloud/core/models.py

```diff
@@ -1246,6 +1251,20 @@ def save(self, recompute_storage=False, *args, **kwargs):
             self.file_storage_bytes = storage.get_project_file_storage_in_bytes(self.id)
         super().save(*args, **kwargs)
 
+    @property
+    def keep_storage(self) -> int:
+        return min(
+            self.storage_keep_versions,
+            self.owner.useraccount.current_subscription.plan.storage_keep_versions,
+        )
+
+    @keep_storage.setter
+    def keep_storage(self, value: Union[int, str]):
+        self.storage_keep_versions = min(
+            value if isinstance(value, int) else int(value),
+            self.owner.useraccount.current_subscription.plan.storage_keep_versions,
+        )
```

Comment:
Personally I would avoid setters in the models as a pattern. However, bigger concern it introduces a new way of doing things in QFC codebase. Similar logic has been already implemented in the `save` model methods in other models.

Aslo, I guess the value can be string when directly input from the UI. IMO the type conversion should happen in the UI responsible for this, I guess a Django (admin?) view.

Reply:
It's possible to avoid using setters. Putting everything inside `save()`.

UI elements from which the input is constructed are HTML <input> fields. I hope the Django template engine does everything automatically.